src/fiber/tests/test_utils.rs

```diff
@@ -272,6 +345,120 @@ impl NetworkNodeConfigBuilder {
     }
 }
 
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn establish_udt_channel_between_nodes(
+    node_a: &mut NetworkNode,
```

Comment:
can we merge this with `establish_channel_between_nodes`? seems only have a extra `udt_script`.

Reply:
I didn't do this because. There are some many places in the code base calling `establish_channel_between_nodes`. This PR is so large that we may encounter a few conflicts when this PR is ready to merge (this happened to the function `create_n_nodes_and_channels_with_index_amounts`).  When this PR is ready to merge, I will do what you said.